a10_neutron_lbaas/db/api.py

```diff
@@ -40,6 +41,25 @@ def get_engine(url=None):
     return sqlalchemy.create_engine(url)
 
 
-def get_session(url=None):
-    DBSession = sqlalchemy.orm.sessionmaker(bind=get_engine())
-    return DBSession()
+def get_session(url=None, **kwargs):
+    DBSession = sqlalchemy.orm.sessionmaker(bind=get_engine(url=url))
+    return DBSession(**kwargs)
+
+
+@contextmanager
+def magic_session(db_session=None, url=None):
```

Comment:
I don't like the name, but we can sort that out later.  Maybe auto_session or context_session.